Sources/Datadog/FeaturesIntegration/WebView/WebRUMEventConsumer.swift

```diff
@@ -6,8 +6,93 @@
 
 import Foundation
 
-internal class WebRUMEventConsumer: WebEventConsumer {
-    func consume(event: [String: Any], eventType: String) {
-        // TODO: RUMM-1791 implement event consumers
+internal class DefaultWebRUMEventConsumer: WebRUMEventConsumer {
+    private let dataWriter: Writer
+    private let dateCorrector: DateCorrectorType
+    private let contextProvider: RUMContextProvider?
+
+    private let jsonDecoder = JSONDecoder()
+
+    init(
+        dataWriter: Writer,
+        dateCorrector: DateCorrectorType,
+        contextProvider: RUMContextProvider?
+    ) {
+        self.dataWriter = dataWriter
+        self.dateCorrector = dateCorrector
+        self.contextProvider = contextProvider
+    }
+
+    func consume(event: JSON) throws {
+        let rumContext = contextProvider?.context
+        let mappedEvent = map(event: event, with: rumContext)
+
+        let jsonData = try JSONSerialization.data(withJSONObject: mappedEvent, options: [])
+        let encodableEvent = try jsonDecoder.decode(CodableValue.self, from: jsonData)
+
+        dataWriter.write(value: encodableEvent)
+    }
+
+    private func map(event: JSON, with context: RUMContext?) -> JSON {
+        guard let context = context,
+              context.sessionID != .nullUUID else {
+            return event
+        }
```

Comment:
_"Given native RUM session rejected by sampler, when receiving event from WebView, it is sent anyway"_ seems to be critical (it impacts user billing) and not very obvious behaviour (in case of RUM disabled we drop events, but we send events no matter of RUM sampling). I don't see it covered in unit tests, meaning we can easily overlook it and change unawarely.